src/admin.rs

```diff
@@ -18,85 +27,71 @@ use node_bindgen::core::JSValue;
 use node_bindgen::core::val::JsObject;
 use node_bindgen::core::buffer::ArrayBuffer;
 
-use crate::SharedScClient;
-
-pub struct AdminScClientWrapper(SharedScClient);
+pub struct FluvioAdminWrapper(SharedFluvio);
 
-impl AdminScClientWrapper {
-    pub fn new(client: SharedScClient) -> Self {
+impl FluvioAdminWrapper {
+    pub fn new(client: SharedFluvio) -> Self {
         Self(client)
     }
 }
 
-
-impl TryIntoJs for AdminScClientWrapper {
+impl TryIntoJs for FluvioAdminWrapper {
     fn try_to_js(self, js_env: &JsEnv) -> Result<napi_value, NjError> {
-        debug!("converting ScClientWrapper to js");
-        let new_instance = AdminScClient::new_instance(js_env, vec![])?;
+        debug!("converting FluvioWrapper to js");
+        let new_instance = FluvioAdminJS::new_instance(js_env, vec![])?;
         debug!("instance created");
-        AdminScClient::unwrap_mut(js_env, new_instance)?.set_client(self.0);
+        FluvioAdminJS::unwrap_mut(js_env, new_instance)?.set_client(self.0);
         Ok(new_instance)
     }
 }
 
-
-pub struct AdminScClient {
-    inner: Option<SharedScClient>,
+pub struct FluvioAdminJS {
+    inner: Option<SharedFluvio>,
 }
 
 #[node_bindgen]
-impl AdminScClient {
+impl FluvioAdminJS {
     #[node_bindgen(constructor)]
     pub fn new() -> Self {
         Self { inner: None }
     }
 
-
-    pub fn set_client(&mut self, client: SharedScClient) {
+    pub fn set_client(&mut self, client: SharedFluvio) {
         self.inner.replace(client);
     }
 
-    
     #[node_bindgen]
-    async fn list_topic(&self) -> Result<ArrayBuffer,ClientError> {
-
-        debug!("list topics");
-
-        let client = self.inner.as_ref().unwrap().clone();
-        let mut client_w = client.write().await;
-
-        let list_topics = client_w.topic_metadata(None).await?;
-
-        let json_slice = serde_json::to_vec(&list_topics)
-        .map_err(|err| ClientError::Other(format!("serialization error: {}",err.to_string())))?;
-
-        // convert to array buffer and wrap in the buffer
-        Ok(ArrayBuffer::new(json_slice))
-
+    async fn list_topic(&self) -> Result<ArrayBuffer, FluvioError> {
+        if let Some(client) = self.inner.clone() {
+            let mut client = client.write().await.admin().await;
+            let topics = client.list::<TopicSpec, _>(vec![]).await?;
+
+            let json_slice = serde_json::to_vec(&topics).map_err(|err| {
+                FluvioError::Other(format!("serialization error: {}", err.to_string()))
+            })?;
+            // // convert to array buffer and wrap in the buffer
+            Ok(ArrayBuffer::new(json_slice))
+        } else {
+            Err(FluvioError::Other("failed to lock client".to_owned()))
```

Comment:
This branch would occur if the client object was not initialized, not if it wasn't locked.